### DISCUSSION TITLE
redirect_uri_mismatch error in production

### DISCUSSION THREAD

**Discussion by AustenLamacraft** [giscus/giscus#509] [Q&A] [answered]
Thanks for giscus, it's really an amazing project!

I have an issue when running giscus (or my own fork) from a development server or one of the temporary domains associated with a Vercel deployment. I get a [redirect_uri_mismatch](https://docs.github.com/en/developers/apps/managing-oauth-apps/troubleshooting-authorization-request-errors) error because the callback URL I've entered in the GitHub settings of my app doesn't match the `redirect_uri`. This is because the `redirect_uri` is formed from an origin that isn't the production deployment.

So I think I understand why this happens, but I'm wondering if there is an elegant way of avoiding this behaviour. Any thoughts would be really helpful! Thanks again!

**Comment by laymonage** [comment, marked as answer]
You can specify multiple callback URLs for your app in this section:

![Add callback URL](https://user-images.githubusercontent.com/6379424/120402124-9f82d280-c36b-11eb-940c-41606361b0d6.png)

See the "Add Callback URL" button on the top right.

You can then specify your production and development endpoints. It doesn't support wildcards, so you'll have to add the URL for every deploy preview provided by Vercel. I think there's a limit on the number of URLs you can add, so remove any deploy previews that are no longer used.

> I'm wondering if there is an elegant way of avoiding this behaviour.

This is the only way that I'm aware of. It's a bit of a hassle, but it's there for security reasons, otherwise any arbitrary URL can get the token once the user has signed in through our app.

**Reply by AustenLamacraft** [reply]
Thanks. I sort of expected this to be the answer but wanted to check.
